### ISSUE TITLE
[JPMS] Accessing resources when Java application run via JavaExec task

### ISSUE THREAD

**Issue by mwkroening** [gradle/gradle#6828]
### Expected Behavior
Accessing Java resources when executing the application via the `run` task should work with the [example configuration](https://guides.gradle.org/building-java-9-modules/#modify_the_code_run_code_task_to_use_modules) of the task.

### Current Behavior
A [MissingResourceException](https://docs.oracle.com/javase/10/docs/api/java/util/MissingResourceException.html) is being thrown when trying to access the resource.

### Context
I am trying to get an application to work as Jigsaw Module. I stumbled about the fact that resources work as expected when the application is built and executed with the start script generated by the [example `startScript` task](https://guides.gradle.org/building-java-9-modules/#modify_the_code_startscript_code_task_for_both_nix_and_windows_to_use_modules), but I can't get resource access to work when using the [example `run` task ](https://guides.gradle.org/building-java-9-modules/#modify_the_code_run_code_task_to_use_modules).

### Steps to Reproduce (for bugs)
I created a simple demo project by copying and pasting [task configurations](https://github.com/mwkroening/modular-gradle-resources/blob/master/build.gradle) from the [Building Java 9 Modules Tutorial](https://guides.gradle.org/building-java-9-modules/):
1. Clone [mwkroening/modular-gradle-resources](https://github.com/mwkroening/modular-gradle-resources)!
2. `./gradlew build` is successful.
3. Run the generated start script in the built distribution package!
    - The access to the ResourceBundle works as expected.
4. Runnning `./gradlew run` fails with a [MissingResourceException](https://docs.oracle.com/javase/10/docs/api/java/util/MissingResourceException.html).

### Your Environment

 * Run scan URL: https://gradle.com/s/3kia3qbspzjoc

**Comment by mwkroening** [comment]
I managed to circumvent the problem by using [gradle-modules-plugin
](https://github.com/java9-modularity/gradle-modules-plugin) as seen in a [seperate branch](https://github.com/mwkroening/modular-gradle-resources/tree/gradle-modules-plugin).

**Comment by jjohannes** [comment]
This is now working in `6.4` (current nightly `gradle-6.4-20200323230320+0000`)

The example can be done like this:

```
plugins {
    id 'application'
}

application {
    mainModule.set('testModule')
    mainClass.set('testPackage.Main')
}

compileJava {
    modularClasspathHandling.getInferModulePath().set(true)
}

run {
    modularClasspathHandling.getInferModulePath().set(true)
}
```

Then the `:run` (JavaExec) task works as expected.

The tart script generation is not yet working like that. This will be fixed in #12613.

**Comment by mwkroening** [comment]
Thanks a lot! I adjusted the example as per your suggestion. That seems to work very well!
